feat(rpc)!: update jsonrpsee to version 0.25.1 and refactor RPC methods

YakupAltay opened the pull request.

YakupAltay commented:
Hey @oblique!

I think there is a problem with the CI. I did not have these errors when I tested locally thanks to your recommendation (https://github.com/eigerco/lumina/issues/575).

I would love to make the necessary changes if you could guide me.

oblique commented:
@YakupAltay The errors from the test cases are a bit weird and my first thought was that a dependency broke deserialization. I saw that you updated all the dependencies instead of just the ones that are related to `jsonrpsee`. After I reverted your `fix: Update Cargo.lock` the tests are passing.

For now revert that commit, run the test and commit only what cargo changed in `Cargo.lock`. Do not update anything else until we find out which dependency breaks our deserialization.

oblique commented:
@YakupAltay I manage to find the culprit. After I update tendermint 0.4.0 to 0.4.3 the tests start to fail. I will investigate it further, but for this PR don't update it.

YakupAltay commented:
Hey @oblique, I reverted it and updated only `jsonrpsee` related packages via this command:
`cargo update -p jsonrpsee`

Yet, the tests are still failing.

YakupAltay commented:
Silly me!

Fixed them all, @oblique.

oblique reviewed (approved):
LGTM! Thanks @YakupAltay

YakupAltay commented:
Fixes #575
